lib/ansible/modules/source_control/gitlab_project_variable.py

```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2019, Markus Bergholz (markuman@gmail.com)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+module: gitlab_project_variable
+short_description: Creates/updates/deletes Gitlab Projects Variables
+description:
+  - When a project variable does not exists, it will be created.
+  - When a project variable does exists, its value will be updated when the values are different.
+  - Variables which are presented in the playbook, but are not presented in the Gitlab project,
+    they stay untouched (purged_vars = False) or will be deleted (purged_vars = True).
+version_added: "2.9"
+author:
+    - "Markus Bergholz (@markuman)"
+requirements:
+  - python >= 2.7
+  - python-gitlab python module
+options:
+  state:
+    description:
+      - create or delete project variable.
+      - Possible values are present and absent.
+    default: present
+    type: str
+    choices: ["present", "absent"]
+  server_url:
+    description:
+      - The URL of the Gitlab server, with protocol (i.e. http or https).
+    required: true
+    type: str
+  login_token:
+    description:
+      - Gitlab access token with api permissions.
+    required: true
+    type: str
+  name:
+    description:
+      - The path and name of the project
+    required: true
+    type: str
+  purge_vars:
+    description:
+      - When set to true, all variables which are not presented in the task will be deleted.
+    default: false
+    required: false
+    type: bool
+  vars:
+    description:
+      - A list of key value pairs
+    default: []
+    required: false
+    type: list
+'''
+
+
+EXAMPLES = '''
+- name: Set or update some CI/CD variables
+  gitlab_project_variables:
+    server_url: https://gitlab.com
+    login_token: secret_access_token
+    name: markuman/dotfiles
+    purge_vars: False
+    vars:
+      - ACCESS_KEY_ID: abc123
+      - SECRET_ACCESS_KEY: 321cba
+
+- name: delete one variable
+  gitlab_project_variables:
+    server_url: https://gitlab.com
+    login_token: secret_access_token
+    name: markuman/dotfiles
+    state: absent
+    vars:
+      - ACCESS_KEY_ID: abc123
+'''
+
+RETURN = '''# '''
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+
+try:
+    import gitlab
+    HAS_GITLAB_PACKAGE = True
+except Exception:
+    HAS_GITLAB_PACKAGE = False
+
+
+class gitlab_project_variables(object):
+
+    def __init__(self, login_token, project_name, server_url):
+        self.repo = gitlab.Gitlab(
+            '{server_url}'.format(server_url=server_url), private_token=login_token)
```

Comment:
What's the point of the `format` here ?

Reply:
https://github.com/markuman/ansible/commit/62ce2a5ac3f9d58a560e903ae8ec195c696e5c17